src/api/reflector.rs

```diff
@@ -132,18 +129,15 @@ where
 
     /// Read data for users of the reflector
     ///
-    /// TODO: deprecate in favour of a stream returning fn..
+    /// TODO: deprecate in favour of a stream returning fn, this would also get rid of the
+    /// intermediate future
     pub fn read(&self) -> Result<Vec<K>> {
-        // unwrap for users because Poison errors are not great to deal with atm.
-        // If a read fails, you've probably failed to parse the Resource into a T
-        // this likely implies versioning issues between:
-        // - your definition of T (in code used to instantiate Reflector)
-        // - current applied kube state (used to parse into T)
-        //
-        // Very little that can be done in this case. Upgrade your app / resource.
-        let cache = self.data.read().unwrap();
-
-        Ok(cache.values().cloned().collect::<Vec<K>>())
+        // Unfortunately we need to spawn and block on a future here
+        // in order to get access to self.data
+        futures::executor::block_on(async {
+            let cache = self.data.lock().await;
+            Ok(cache.values().cloned().collect::<Vec<K>>())
+        })
```

Comment:
Will turn this into an `async fn data(&self) -> Result<Vec<K>>` soon.

If people have to call this from their main business logic, and it can block, we should expose the future.